include/libtorrent/aux_/session_impl.hpp

```diff
@@ -180,7 +176,10 @@ namespace libtorrent
 		struct tracker_logger;
 #endif
 
-		TORRENT_EXPORT std::pair<bencode_map_entry*, int> settings_map();
+#ifndef TORRENT_DISABLE_DHT
+		TORRENT_EXTRA_EXPORT dht_settings read_dht_settings(bdecode_node const& e);
```

Comment:
I would have thought these two functions should live under ``src/kademlia``. they could still be made public here with a ``using`` statement.